PayloadPkg/FirmwareUpdate/FirmwareUpdate.c

```diff
@@ -863,18 +856,26 @@ ApplyFwImage (
     OUT BOOLEAN                       *ResetRequired
   )
 {
-  EFI_STATUS  Status;
-  VOID        *CsmeUpdateInData;
+  EFI_STATUS      Status;
+  UINT32          Signature;
+  VOID            *CsmeUpdateInData;
   BOOT_PARTITION  Partition;
 
   Status = EFI_SUCCESS;
   *ResetRequired = FALSE;
 
-  if (CompareGuid(&ImageHdr->UpdateImageTypeId, &gSblFWUpdateImageFileGuid)) {
-    Status = UpdateSystemFirmware(ImageHdr);
-  } else if (CompareGuid(&ImageHdr->UpdateImageTypeId, &gCfgFWUpdateImageFileGuid)) {
+  if (ImageHdr == NULL) {
+    return EFI_NOT_FOUND;
+  }
+
+  Signature = (UINT32)ImageHdr->UpdateHardwareInstance;

```

Comment:
Later we could think about to optimize capsule header to use component signature directly.

Reply:
we might still need the GUID for ESRT compatibility

Reply:
It is a little confusing for the name UpdateHardwareInstance assigned to Signature. 
Not sure if there is any benefit for ESRT compatibility.  we could think about it later.